avahi-daemon/simple-protocol.c

```diff
@@ -276,7 +277,7 @@ static void handle_line(Client *c, const char *s) {
     if (c->state != CLIENT_IDLE)
         return;
 
-    if ((n_args = sscanf(s, "%63s %63s", cmd, arg)) < 1 ) {
+    if ((n_args = sscanf(s, "%63s %1013s", cmd, arg)) < 1 ) {
```

Comment:
I don't like not using constant here, but not sure how to get length here. Without using completely different calls.

Reply:
I don't like it either but given that it's unlikely to ever change I'd just use
```c
char cmd[64], arg[1014];
```
instead to keep it simple and somewhat consistent with the hard-coded `cmd[64]`. Could you also add something like
```diff
diff --git a/.github/workflows/smoke-tests.sh b/.github/workflows/smoke-tests.sh
index b550a04..dcfc499 100755
--- a/.github/workflows/smoke-tests.sh
+++ b/.github/workflows/smoke-tests.sh
@@ -58,6 +58,7 @@ avahi-resolve -v -a "$ipv6addr"
 cmds=(
     "HELP"
     "RESOLVE-HOSTNAME $h"
+    "RESOLVE-HOSTNAME $(perl -le '(print q/A/ x 1014)')"
     "RESOLVE-HOSTNAME-IPV6 $h"
     "RESOLVE-HOSTNAME-IPV4 $h"
     "RESOLVE-ADDRESS $ipv4addr"
```
to the smoke test? Even though `radamsa` generates all sorts of weird things I think it would be better to test this hard-coded assumption explicitly too. Once the CI with that test passes feel free to merge the PR.

Reply:
Let's keep AVAHI_DOMAIN_NAME_MAX here. Looks like this pattern is used in other places too. Ideally it should be replaced but it can be done later.

I'll add the test in a separate PR.